pkg/3scale/amp/operator/redis_options_provider.go

```diff
@@ -51,13 +58,127 @@ func (r *RedisOptionsProvider) GetRedisOptions() (*component.RedisOptions, error
 
 	r.setPersistentVolumeClaimOptions()
 
-	err := r.options.Validate()
+	// Should the operator be reading redis secrets?
+	// When HA is disabled, do we support external redis?
+	// If answer is true, why does the operator deploy redis?
+	// If the answer is no, then it would be sufficient to set default URL's (internal redis url)
```

Comment:
See my comment https://github.com/3scale/3scale-operator/pull/486/files#r500176502.

Even if the answer is no I think we should reconcile other fields to allow non-default values for fields of the secret.
This applies to system redis and to system database too.

Reply:
When redis is deployed internally, sentinel configuration and namespace conf does not have any sense. 

This change may belong to another PR, but I have my doubts about the current implementation, hence bringing up the discussion.

Reply:
sentinel configuration does not make sense indeed.
namespace conf I think be configured without issues.
Some secret attributes make sense when redis DBs are internal and some don't as commented in https://github.com/3scale/3scale-operator/pull/486#discussion_r500177300